src/sql/workbench/contrib/queryplan2/browser/queryPlanWidgetBase.ts

```diff
@@ -0,0 +1,25 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export abstract class QueryPlanWidgetBase {
+	public container: HTMLElement;
+	public identifier: string;
+
+	/**
+	 *
+	 * @param container HTML Element that contains the UI for the plan action view.
+	 * @param identifier Uniquely identify the view to be added or removed. Note: Only 1 view with the same id can be added to the controller
+	 */
+	constructor(container: HTMLElement, identifier: string) {
+		this.container = container;
```

Comment:
Just put the modifiers in the constructor 

`constructor(public container: HTMLElement, public identifier: string) {`

Should these be readonly?